app/queues/common/expiration-request.server.ts

```diff
@@ -0,0 +1,100 @@
+import { Worker, Queue, UnrecoverableError } from 'bullmq';
+
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { prisma } from '~/db.server';
+
+import { addNotification } from '../notifications/notifications.server';
+import { deleteDnsRequest } from '../dns/delete-record-flow.server';
+import { RecordType } from '../notifications/expiration-notification.server';
+
+import type { ExpirationStatusPayload as ExpirationRequestPayload } from '../notifications/expiration-notification.server';
+
+const { EXPIRATION_REPEAT_FREQUENCY, JOB_REMOVAL_FREQUENCY } = process.env;
+const JOB_REMOVAL_INTERVAL = 7 * 24 * 60 * 60 * 1000;
+declare global {
+  var __expiration_request_init__: boolean;
+}
+// queue name
+const expirationRequestQueueName = 'expiration-request';
+
+// query to fetch expired records from database based on type
+const getExpiredRecords = (type: RecordType) => {
+  switch (type) {
+    case RecordType.DnsRecord:
+      return prisma.record.findMany({
+        where: {
+          expiresAt: {
+            lt: new Date(),
+          },
+        },
+        include: {
+          user: true,
+        },
+      });
+    case RecordType.Certificate:
+      logger.info(`TODO: return expired certificates from the db`);
+      break;
+    default:
+      throw new Error(`Invalid recordType: ${type}`);
+  }
+};
+
+// queue initialization
+const expirationRequestQueue = new Queue<ExpirationRequestPayload>(expirationRequestQueueName, {
+  connection: redis,
+});
+
+export function addExpirationRequest(type: RecordType) {
+  const jobName = `expirationRequestQueueName-${type}`;
+  return expirationRequestQueue.add(
+    jobName,
+    { type },
+    {
+      repeat: { every: Number(EXPIRATION_REPEAT_FREQUENCY) || 24 * 60 * 60 * 1000 },
```

Comment:
I don't think you can use `Number()` here, since this is a cron string, right?

Reply:
We are not using the cron pattern anymore since not everyone is familiar with it. Therefore, we are casting the value of the environment variable into a Number, as "every" accepts a value in milliseconds.

Reply:
Above you have this:

```ini
# Background jobs
EXPIRATION_REPEAT_FREQUENCY=24 * 60 * 60 * 1000
JOB_REMOVAL_FREQUENCY=7 * 24 * 60  * 60 * 1000
```

Should you change that if you're not using CRON any more?  Doing `Number('7 * 24 * 60  * 60 * 1000')` will give `NaN`.

Reply:
Right, the quotes.